androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java

```diff
@@ -81,59 +138,471 @@ protected Intent getActivityIntent() {
     };
 
     @Before
-    public void init(){
-        UiDevice uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
-        /*Point[] coordinates = new Point[4];
-        coordinates[0] = new Point(248, 1020);
-        coordinates[1] = new Point(248, 429);
-        coordinates[2] = new Point(796, 1020);
-        coordinates[3] = new Point(796, 429);*/
-        try {
-            if (!uiDevice.isScreenOn()) {
-                uiDevice.wakeUp();
-                //uiDevice.swipe(coordinates, 10);
+    public void init() {
+        Bundle arguments = InstrumentationRegistry.getArguments();
+
+        testUser = arguments.getString("TEST_USER");
+        testUser2 = arguments.getString("TEST_USER2");
+        testPassword = arguments.getString("TEST_PASSWORD");
+        testPassword2 = arguments.getString("TEST_PASSWORD2");
+        testServerURL = arguments.getString("TEST_SERVER_URL");
+        trusted = Integer.parseInt(arguments.getString("TRUSTED"));
+        switch (trusted) {
```

Comment:
I would move this logic to get an enum from an int to a static method in the enum, similar to the method ```fromValue``` here: https://github.com/owncloud/android/blob/master/src/com/owncloud/android/datamodel/UploadsStorageManager.java#L76

That way you could read the servertype here, or at any other place, directly as

```servertype = ServerType.fromValue(Integer.parseInt(arguments.getString("TRUSTED")));```

and the field ```trusted``` wouldn't be needed anymore.